modules/core/src/smithy4s/http/HttpUri.scala

```diff
@@ -18,7 +18,7 @@ package smithy4s.http
 
 final case class HttpUri(
```

Comment:
**suggestion:** we could safeguard against bincompat breakage in the future, now that we're at it:

https://docs.scala-lang.org/overviews/core/binary-compatibility-for-library-authors.html

this would mean e.g. hiding the `copy` method, and opting for `.withXXX(newXXX)` methods instead, and would need to touch more than just this type though.

cc @Baccata

Reply:
Good point :+1:

Reply:
Do you mind if I make a dedicated issue for that so we can track it for the whole of smithy4s?
It could be an important part of a 0.19 update

Reply:
Sure. In that issue, you should add the presence of a dedicated scalafix rule aimed at finding such cases.

Reply:
Ok, I'll create the issue and look into it

Reply:
https://github.com/disneystreaming/smithy4s/issues/1349